iati/core/tests/test_rulesets.py

```diff
@@ -162,57 +162,67 @@ def test_rule_init_no_parameters(self, rule_constructor):
     def test_rule_init_invalid_case(self, rule_constructor, case):
         """Check that a Rule cannot be created when case is not a dictionary."""
-        xpath_base = 'an xpath'
+        context = 'an xpath'
 
         with pytest.raises(ValueError):
-            rule_constructor(xpath_base, case)
+            rule_constructor(context, case)
 
     def test_rule_init_invalid_case_property(self, rule_constructor):
         """Check that a Rule cannot be created when a case has a property that is not permitted."""
-        xpath_base = 'an xpath'
+        context = 'an xpath'
         case = {'thisis_an_invalidkey': ['this_is_a_value']}
 
         with pytest.raises(ValueError):
-            rule_constructor(xpath_base, case)
+            rule_constructor(context, case)
 
 
 class RuleSubclassTestBase(object):
     """A base class for Rule subclass tests."""
 
     @pytest.fixture
-    def xpath_base(self):
-        """Return valid xpath_base."""
+    def valid_single_context(self):
+        """Return valid context with a single match."""
         return '//root_element'
 
     @pytest.fixture
-    def empty_xpath_base(self):
-        """Empty_xpath_base."""
-        return ''
+    def valid_multiple_context(self):
+        """Return a valid context with multiple matches."""
+        return '//nest'
 
-    @pytest.fixture
-    def rule_basic_init(self, rule_constructor, instantiating_case, xpath_base):
-        """Rule subclass."""
-        return rule_constructor(xpath_base, instantiating_case)
+    @pytest.fixture(params=[
+        'count(condition)>0',
+        'condition'
+    ])
+    def valid_condition_case(self, validating_case, request):
+        """Return a case with optional condition attribute."""
+        condition_validating_case = deepcopy(validating_case)
+        condition_validating_case['condition'] = request.param
+        return condition_validating_case
 
-    # @pytest.fixture
-    # def rule_is_valid_for_case(self, rule_constructor, xpath_base, case_for_is_valid_for):
-    #     """Rule with specific cases for checking the `is_valid_for` function."""
-    #     return rule_constructor(xpath_base, case_for_is_valid_for)
+    @pytest.fixture(params=[
+        'count(condition)>0',
+        'condition'
+    ])
+    def invalid_condition_case(self, invalidating_case, request):
+        """Return a case with optional condition attribute."""
+        condition_invalidating_case = deepcopy(invalidating_case)
+        condition_invalidating_case['condition'] = request.param
+        return condition_invalidating_case
 
     @pytest.fixture
-    def rule_empty_xpath_base(self, rule_constructor, instantiating_case, empty_xpath_base):
-        """Rule subclass with an empty xpath_base."""
-        return rule_constructor(empty_xpath_base, instantiating_case)
+    def rule_basic_init(self, rule_constructor, instantiating_case, valid_single_context):
```

Comment:
Suggest renaming the various Rule fixtures to `rule_instantiating`, `rule_valid` and `rule_invalid` - the current names don't quite flow in the various places that they're used.

Reply:
done